syntax = "proto3";

// 计算数据的摘要
//      * 低优先级服务，可用软算法替代
//      * 允许客户端使用流多次传入数据
//      * 服务端在客户端 complete 之后返回最后的计算值
service DigestService {
    rpc Digest (stream DigestRequest) returns (DigestResponse) {}
}

// transformation 摘要算法名称，这里不使用枚举是便于在不改动服务的情况下兼容新的 hash 算法
//      * 算法名称大小写不敏感
//      * 兼容算法常见的别名
//      * 支持的算法必须包括
//          - MD5
//          - SHA SHA1 SHA-1
//          - SHA224 SHA-224
//          - SHA2 SHA-2 SHA256 SHA-256
//          - SHA384 SHA-384
//          - SHA512 SHA-512
//
// messages 用于计算摘要的数据
//      * repeated 在于方便需要计算多组数据拼接值的摘要时，不用在客户端手动拼接数据
message DigestRequest {
    string transformation = 1;
    repeated bytes messages = 2;
}

message DigestResponse {
    bytes digest = 1;   // 数据的摘要值
}



